sdk/compiler/damlc/daml-stdlib-src/DA/CCTP.daml

```diff
@@ -0,0 +1,79 @@
+-- Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- @SINCE-LF 2.dev
+
+{-# LANGUAGE CPP #-}
+
+-- | Functions for working with CCTP.
+module DA.CCTP
+  (keccak256
+  , secp256k1
+  , BytesHex
+  , Bytes32Hex
+  , UInt32Hex
+  , UInt64Hex
+  , UInt256Hex
+  , PublicKeyHex
+  , SignatureHex
+  ) where
+
+import DA.Text qualified as Text
+import GHC.Types (primitive)
+
+-- TODO: https://github.com/DACH-NY/canton-network-utilities/issues/2943: Introduce ByteString opaque data types
+
+type BytesHex = Text
+type Bytes32Hex = Text
+type UInt32Hex = Text
+type UInt64Hex = Text
+type UInt256Hex = Text
+
+-- | A DER formatted public key to be used for ECDSA signature verification
+type PublicKeyHex = Text
+
+-- | A DER formatted SECP256K1 signature
+type SignatureHex = Text
+
+hexByteToInt : Text -> Int
+hexByteToInt "0" = 0
+hexByteToInt "1" = 1
+hexByteToInt "2" = 2
+hexByteToInt "3" = 3
+hexByteToInt "4" = 4
+hexByteToInt "5" = 5
+hexByteToInt "6" = 6
+hexByteToInt "7" = 7
+hexByteToInt "8" = 8
+hexByteToInt "9" = 9
+hexByteToInt "a" = 10
+hexByteToInt "b" = 11
+hexByteToInt "c" = 12
+hexByteToInt "d" = 13
+hexByteToInt "e" = 14
+hexByteToInt "f" = 15
+hexByteToInt "A" = 10
+hexByteToInt "B" = 11
+hexByteToInt "C" = 12
+hexByteToInt "D" = 13
+hexByteToInt "E" = 14
+hexByteToInt "F" = 15
+hexByteToInt hexChars | Text.length hexChars == 2 && Text.isHex hexChars = (hexByteToInt (Text.substring 1 1 hexChars)) + 16 * (hexByteToInt (Text.substring 0 1 hexChars))
+hexByteToInt hexChars | not (Text.isHex hexChars) = error $ "Argument is not a hex string: " <> hexChars
+hexByteToInt hexChars = error $ "Hex string should have length <= 2: " <> hexChars
```

Comment:
`hexByteToInt` could reside in `Text.daml`, but I felt it didn't really belong there?